etna/auto/auto.py

```diff
@@ -37,7 +39,152 @@ def __call__(self, pipeline: Pipeline) -> None:
         ...
 
 
-class Auto:
+class AutoBase:
+    """Base Class for Auto and Tune, implementing core logic behind these classes."""
+
+    def __init__(self, target_metric: Metric, metric_aggregation: MetricAggregationStatistics = "mean"):
+        # this code is never executed, its single purpose is to help linter (and user reading code) infer parameters types
+        self.target_metric: Metric = target_metric
+        self.metric_aggregation: MetricAggregationStatistics = metric_aggregation
+        self._optuna: Optional[Optuna] = None
+
+    def summary(self) -> pd.DataFrame:
+        """Get Auto trials summary."""
+        if self._optuna is None:
+            self._optuna = self._init_optuna()
+
+        study = self._optuna.study.get_trials()
+
+        study_params = [
+            {**trial.user_attrs, "pipeline": get_from_params(**trial.user_attrs["pipeline"]), "state": trial.state}
+            for trial in study
+        ]
+
+        return pd.DataFrame(study_params)
+
+    def top_k(self, k: int = 5) -> List[Pipeline]:
+        """
+        Get top k pipelines.
+
+        Parameters
+        ----------
+        k:
+            number of pipelines to return
+        """
+        summary = self.summary()
+        df = summary.sort_values(
+            by=[f"{self.target_metric.name}_{self.metric_aggregation}"],
+            ascending=(not self.target_metric.greater_is_better),
+        )
+        return [pipeline for pipeline in df["pipeline"].values[:k]]  # noqa: C416
+
+    def _init_optuna(self):
```

Comment:
This method should stay in `Auto` class. According to #1023 we want only `fit`, `summary`, `top_k` to be implemented in `AutoBase`.